nhrpd/vici.c

```diff
@@ -469,16 +469,49 @@ static void vici_register_event(struct vici_conn *vici, const char *name)
 	vici_submit(vici, obuf);
 }
 
+static bool vici_charon_filepath_done;
+
+static char *vici_get_charon_filepath(void)
+{
+	static char buff[1200];
+	FILE *fp;
+	char *ptr = buff, *ptr2;
+	char line[1024];
+
+	if (vici_charon_filepath_done)
+		return ptr;
+	fp = popen("ipsec --piddir", "r");
+	if (!fp) {
+		zlog_err("%% Failed to retrieve charon file path.");
```

Comment:
log - what could be done if this appears (every couple of seconds) ?

Reply:
I remember of an algorithm for backoff logs, but I did not find any piece of code for that.
instead I propose to use a boolean to prevent from printing error message.
also, I use flog_err api instead of zlog_err.